src/test/AludelV3.t.sol

```diff
@@ -296,4 +270,359 @@ contract AludelV3Test is Test {
+
+    AludelV3.AludelInitializationParams private defaultInitParams;
+    Utils.LaunchParams private defaultLaunchParams;
+
+    event AludelFunded(uint256 amount, uint256 duration);
+
+    function setUp() public {
+        UserFactory userFactory = new UserFactory();
+        user = userFactory.createUser("user", 0);
+        anotherUser = userFactory.createUser("anotherUser", 1);
+        admin = userFactory.createUser("admin", 2);
+        recipient = userFactory.createUser("recipient", 3);
+
+        Crucible crucibleTemplate = new Crucible();
+        crucibleTemplate.initializeLock();
+        crucibleFactory = new CrucibleFactory(address(crucibleTemplate));
+
+        // feeBps set to 0 so accounting calculations are easier to comprehend
+        bps = 0;
+        factory = new AludelFactory(recipient.addr(), bps);
+
+        AludelV3 template = new AludelV3();
+        template.initializeLock();
+
+        rewardPoolFactory = new RewardPoolFactory();
+        powerSwitchFactory = new PowerSwitchFactory();
+        stakingToken = new MockERC20("", "STK");
+        rewardToken = new MockERC20("", "RWD");
+
+        rewardScaling = IAludelV3.RewardScaling({floor: 1, ceiling: 1, time: 1 days});
+
+        factory.addTemplate(address(template), "aludel v3", false);
+
+        defaultInitParams = AludelV3.AludelInitializationParams({
+            rewardPoolFactory: address(rewardPoolFactory),
+            powerSwitchFactory: address(powerSwitchFactory),
+            stakingToken: address(stakingToken),
+            rewardToken: address(rewardToken),
+            rewardScaling: rewardScaling,
+            hookContract: IAludelHooks(address(0))
+        });
+
+        defaultLaunchParams = Utils.LaunchParams({
+            template: address(template),
+            name: "name",
+            stakingTokenUrl: "https://staking.token",
+            startTime: START_TIME,
+            vaultFactory: address(crucibleFactory),
+            bonusTokens: new address[](0),
+            owner: admin.addr(),
+            initParams: abi.encode(defaultInitParams)
+        });
+
+        aludel = AludelV3(Utils.launchProgram(factory, defaultLaunchParams));
+        vm.warp(START_TIME);
+    }
+
+    modifier callerOwner() {
+        vm.startPrank(admin.addr());
+        _;
+    }
+
+    function test_revert_when_caller_is_not_owner(uint256 reward, uint256 duration) public {
+        reward = bound(reward, 1, MAX_REWARD);
+        duration = bound(duration, 1, 1000 days);
+
+        Utils.fundMockToken(admin.addr(), rewardToken, uint256(reward));
+
+        // start prank as user (not the owner)
+        vm.startPrank(user.addr());
+
+        rewardToken.approve(address(aludel), uint256(reward));
+
+        vm.expectRevert("Ownable: caller is not the owner");
+        aludel.fund(reward, duration);
+    }
+
+    function test_revert_when_aludel_has_not_enough_allowance(uint256 reward, uint256 duration, uint256 allowance)
```

Comment:
these tests are nice to have but they aren't really related to the cleanup of reward schedules. They could've been done in a separate PR and keep things a bit tidier

Reply:
Although I understand with what you're saying, I don't see the problem really.
Actually in this case if I had to create another branch just to add certain tests it would just killed my workflow.
I would agree with you if instead of tests these changes were in the implementation, but because the cleanup is only executed when `fund` is called I would even argue that making these changes here is desirable.

I can compromise that to keep it strictly separate in the future.

Reply:
In the future what you could do is, when you realize you're making changes that don't fit in the current PR but are nevertheless worthwhile:

- get the changes into a separate commit
- get the commit out of the current branch
- cherrypick the commit into a separate branch later